packages/serverpod_serialization/lib/src/copy_with.dart

```diff
@@ -68,13 +53,14 @@ dynamic _guardedCopyWith(dynamic element) {
   if (element is Enum) return element;
 
   // Required as the extension with clone() is not found otherwise.
-  if (element is ByteData) return element.clone();
-  if (element is List) return element.clone();
-  if (element is Map) return element.clone();
+  if (element is ByteData) return element.clone() as T;
+  if (element is List || element is Map) {
+    throw Exception("Can't clone `List` or `Map` types");
+  }
 
   try {
-    return element.copyWith();
+    return (element as dynamic).copyWith() as T;
   } on NoSuchMethodError {
-    throw 'No copyWith method found on ${element.runtimeType}';
+    throw Exception('No copyWith method found on ${element.runtimeType}');
```

Comment:
@SandPod Would you like me to throw `NoSuchMethodError` here instead of `Exception`?

Reply:
The suggested solution in the PR is good now! 👍